docs/proposal-007-instance-selector.md

```diff
@@ -0,0 +1,269 @@
+# EC2 Instance Selector Support for Nodegroups
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+- [Design](#design)
+  - [eksctl create cluster](#eksctl-create-cluster)
+  - [eksctl create nodegroup](#eksctl-create-nodegroup)
+  - [API Changes](#api-changes)
+  - [Caveats](#caveats)
+<!-- /toc -->
+
+## Summary
+
+This proposal adds support for the EC2 instance selector for both managed and unmanaged nodegroups.
+
+
+## Motivation
+
+With support for multiple instance types and Spot instances for EKS Managed Nodegroups, users can create Spot instances where EKS will configure and launch an Autoscaling Group of Spot instances following  Spot best practices. Users can set `managedNodeGroup.spot` and supply a list of instance types in `managedNodeGroup.instanceTypes` to instruct eksctl to create a managed nodegroup using Spot instances.
+
+It is a Spot best practice to use multiple instance types, but with over 270 EC2 instance types, users have to spend time figuring out which instance types would be well suited for their nodegroup.
+
+The [EC2 Instance Selector](https://github.com/aws/amazon-ec2-instance-selector) tool addresses this problem by generating a list of instance types based on resource criteria such as vCPUs, memory, etc. This proposal aims to integrate the EC2 instance selector with eksctl to enable creating nodegroups with multiple instance types by passing the resource criteria.
+
+### Problem: a simple integration
+We could add the instance selector options—as CLI options (`--instance-selector-vcpus` and `--instance-selector-memory`) and in the config file—to `create cluster` and `create nodegroup` that would select the instance types matching the resource criteria for the user and proceed to creating the nodegroup. However, this gives users no control over selecting the instance types. Users may want to inspect and change the instances matching the instance selector criteria before proceeding to creating a nodegroup, and also to document the instance types being used for their nodegroups in the ClusterConfig file.
```

Comment:
This is a strawman against a certain option. Can we be more clear about the problem we want to solve here?

In particular, is it critical that we have a "human in the loop" here? We could always modify the config file with the selections and output them. Is there benefit to having the manual selection process? Doesn't this require the user to have an understanding of the instance type options anyways?

Reply:
Reading further - it looks like this is actually the proposal - but we negate it early to introduce the `--dry-run` functionality. Let's consider cutting or revising this section as its confusing.